ingestr/main_test.py

```diff
@@ -1538,3 +1545,188 @@ def test_db_to_db_exclude_columns(source, dest):
+        ],
+        "ProvisionedThroughput": {
+            "ReadCapacityUnits": 35000,
+            "WriteCapacityUnits": 35000,
+        },
+    }
+
+    items = [
+        {"id": {"S": "1"}, "updated_at": {"S": "2024-01-01T00:00:00"}},
+        {"id": {"S": "2"}, "updated_at": {"S": "2024-02-01T00:00:00"}},
+        {"id": {"S": "3"}, "updated_at": {"S": "2024-03-01T00:00:00"}},
+    ]
+
+    def load_test_data(ls):
+        client = ls.get_client("dynamodb")
+        client.create_table(**table_cfg)
+        for item in items:
+            client.put_item(TableName=db_name, Item=item)
+    
+    def items_to_list(items):
+        """converts dynamodb item list to list of dics"""
+        result = []
+        for i in items:
+            entry = {}
+            for key, val in i.items():
+                entry[key] = list(val.values())[0]
+            result.append(entry)
+        return result
+
+    local_stack = LocalStackContainer(
+        image="localstack/localstack:4.0.3"
+    ).with_services("dynamodb")
+    local_stack.start()
+    wait_for_logs(local_stack, "Ready.")
+    load_test_data(local_stack)
+
+    dynamodb_url = urlparse(local_stack.get_url())
+    src_uri = (
+        f"dynamodb://{dynamodb_url.netloc}?" +
+        f"region={local_stack.env['AWS_DEFAULT_REGION']}&" +
+        f"access_key_id={local_stack.env['AWS_ACCESS_KEY_ID']}&" +
+        f"secret_access_key={local_stack.env['AWS_SECRET_ACCESS_KEY']}"
+    )
+    yield DynamoDBTestConfig(
+        db_name,
+        src_uri,
+        items_to_list(items),
+    )
+
+    local_stack.stop()
+
+
+def dynamodb_tests() -> Iterable[Callable]:
+    def assert_success(result):
+        if result.exception is not None:
+            traceback.print_exception(*result.exc_info)
+            raise AssertionError(result.exception)
+
+
+    def smoke_test(dest, dynamodb):
+        dest_uri = dest.start()
+        dest_table = f"public.dynamodb_{get_random_string(5)}"
+        dest_engine = sqlalchemy.create_engine(dest_uri)
+
+        result = invoke_ingest_command(
+            dynamodb.uri,
+            dynamodb.db_name,
+            dest_uri,
+            dest_table,
+            "append",
+            "updated_at"
+        )
+
+        assert_success(result)
+        result = dest_engine.execute(f"SELECT id, updated_at from {dest_table} ORDER BY id").fetchall()
+        assert len(result) == 3
+        for i in range(len(result)):
+            assert result[i][0] == dynamodb.data[i]["id"]
+            assert result[i][1] == pendulum.parse(dynamodb.data[i]["updated_at"])
+            
+        dest.stop()
+
+    def append_test(dest, dynamodb):
+        dest_uri = dest.start()
+        dest_table = f"public.dynamodb_{get_random_string(5)}"
+
+        # connection pooling causes issues with duckdb, when the connection
+        # is reused below, so we disable pooling.
+        dest_engine = sqlalchemy.create_engine(dest_uri, poolclass=NullPool)
```

Comment:
this is pretty clever, I had to do an uglier solution before, didn't know I could do this 👍